src/System.Globalization/tests/DateTimeFormatInfo/DateTimeFormatInfoData.cs

```diff
@@ -41,11 +44,144 @@ public static string GetAbbreviatedEraName(CultureInfo cultureInfo)
+        {
+            if (string.Equals(cultureInfo.Name, "en-US", StringComparison.OrdinalIgnoreCase))
+            {
+                return new string[]
+                {
+                    "Sun",
+                    "Mon",
+                    "Tue",
+                    "Wed",
+                    "Thu",
+                    "Fri",
+                    "Sat"
+                };
+            }
+            if (string.Equals(cultureInfo.Name, "fr-FR", StringComparison.OrdinalIgnoreCase))
+            {
+                string[] dayNames = new string[]
+                {
+                    "dim.",
+                    "lun.",
+                    "mar.",
+                    "mer.",
+                    "jeu.",
+                    "ven.",
+                    "sam."
+                };
+
+                if (s_isOSX)
+                {
+                    CapitalizeStrings(dayNames);
+                }
+                return dayNames;
+            }
+
+            throw GetCultureNotSupportedException(cultureInfo);
+        }
+
+        private static void CapitalizeStrings(string[] strings)
+        {
+            for (int i = 0; i < strings.Length; i++)
+            {
+                strings[i] = strings[i].Substring(0, 1).ToUpper() + strings[i].Substring(1);
+            }
+        }
+
+        internal static CalendarWeekRule GetCalendarWeekRule(CultureInfo cultureInfo)
+        {
+            if (string.Equals(cultureInfo.Name, "en-US", StringComparison.OrdinalIgnoreCase))
+            {
+                return CalendarWeekRule.FirstDay;
+            }
+            if (string.Equals(cultureInfo.Name, "br-FR", StringComparison.OrdinalIgnoreCase))
+            {
+                if (s_isWindows && s_WindowsVersion < 10)
+                {
+                    return CalendarWeekRule.FirstFullWeek;
+                }
+                else
+                {
+                    return CalendarWeekRule.FirstFourDayWeek;
+                }
+            }
+
+            throw GetCultureNotSupportedException(cultureInfo);
+        }
+
+        public static Exception GetCultureNotSupportedException(CultureInfo cultureInfo)
+        {
+            return new NotSupportedException(string.Format("The culture '{0}' with calendar '{1}' is not supported.",
+                cultureInfo.Name,
                 cultureInfo.Calendar.GetType().Name));
         }
+
+        public static int GetWindowsVersion()
+        {
+            if (s_isWindows)
+            {
+                RTL_OSVERSIONINFOEX osvi = new RTL_OSVERSIONINFOEX();
+                osvi.dwOSVersionInfoSize = (uint)Marshal.SizeOf(osvi);
+                Assert.Equal(0, RtlGetVersion(out osvi));
+                return (int)osvi.dwMajorVersion;
+            }
+
+            return -1;
+        }
+
+        [DllImport("ntdll.dll")]
+        private static extern int RtlGetVersion(out RTL_OSVERSIONINFOEX lpVersionInformation);
```

Comment:
I think this API intended to be kernel mode and it is not nice to call it from the user mode even if it is working. I would suggest you use IsWindowsVersionOrGreater instead